grammars/javascript.cson

```diff
@@ -1251,3 +1322,82 @@
         ]
       }
     ]
+  'switch_statement':
+    'patterns': [
+      {
+        'begin': '\\b(switch)\\b'
+        'beginCaptures':
+          '0':
+            'name': 'keyword.control.switch.js'
+        'end': '}'
+        'endCaptures':
+          '0':
+            'name': 'punctuation.definition.section.switch-block.end.js'
+        'patterns': [
+          {
+            'begin': '\\('
+            'beginCaptures':
+              '0':
+                'name': 'punctuation.definition.switch-expression.begin.js'
+            'end': '\\)'
+            'endCaptures':
+              '0':
+                'name': 'punctuation.definition.switch-expression.end.js'
+            'patterns': [
+              'include': '$self'
+            ]
+          }
+          {
+            'begin': '{'
+            'beginCaptures':
+              '0':
+                'name': 'punctuation.definition.section.switch-block.begin.js'
+            'end': '(?=})'
+            'patterns': [
+              {
+                'begin': '\\b(case)\\b'
+                'end': ':'
+                'beginCaptures':
+                  '0':
+                    'name': 'keyword.control.case.js'
+                'endCaptures':
+                  '0':
+                    'name': 'punctuation.definition.section.case-statements.js'
+                'patterns': [
+                  {
+                    'match': '(\\w+)(?=\\s*:)'
+                    'captures':
+                      '1':
+                        'patterns': [
+                          'include': '$self'
+                        ]
+                  }
+                  {
+                    'match': '(?:".*")|(?:\'.*\')(?=\\s*:\\s*\\.*)'
+                    'captures':
+                      '0':
+                        'patterns': [
+                          'include': '#strings'
+                        ]
+                  }
+                  {
+                    'include': '$self'
+                  }
+                ]
+              }
+              {
+                'match': '(?:^\\s*)?\\b(default)\\b\\s*(:)'
+                'captures':
+                  '1':
+                    'name': 'keyword.control.default.js'
+                  '2':
+                    'name': 'punctuation.definition.section.case-statements.js'
```

Comment:
Same, I think this should probably be either `case-statement` or `default-statement`